### PR TITLE
fix(gatsby,gatsby-plugin-image): fix createRoot on React 18

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Move to `createRoot` for gatsby-plugin-image and indicator

## Related Issues

https://github.com/gatsbyjs/gatsby/discussions/31943

### CODE REVIEW

File: packages/gatsby/cache-dir/app.js [hunk 1]
```diff
@@ -194,7 +194,7 @@ apiRunnerAsync(`onClientEntry`).then(() => {
         document.body.append(indicatorMountElement)
 
         if (renderer === ReactDOM.hydrateRoot) {
-          renderer(indicatorMountElement).render(
+          ReactDOM.createRoot(indicatorMountElement).render(
```
Comment (wardpeet) [inline packages/gatsby/cache-dir/app.js:197]: There is not html to hydrate so we d a regular root

File: packages/gatsby-plugin-image/src/components/lazy-hydrate.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import React, { MutableRefObject } from "react"
-import { hydrate, render } from "react-dom"
+// @ts-ignore - react 18 has hydrateRoot
+import { hydrate, render, createRoot } from "react-dom"
```
Comment (LekoArts) [inline packages/gatsby-plugin-image/src/components/lazy-hydrate.tsx:3]: Did you mean to use `hydrateRoot` or is the ts-ignore comment off?

File: packages/gatsby-plugin-image/src/components/lazy-hydrate.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React, { MutableRefObject } from "react"
-import { hydrate, render } from "react-dom"
+// @ts-ignore - react 18 has hydrateRoot
```
Comment (wardpeet) [inline packages/gatsby-plugin-image/src/components/lazy-hydrate.tsx:2]:
```suggestion
// @ts-ignore - react 18 has createRoot
```

File: packages/gatsby-plugin-image/src/components/lazy-hydrate.tsx (hunk 2 above)
Comment (wardpeet) [reply packages/gatsby-plugin-image/src/components/lazy-hydrate.tsx:3]: createRoot

**Review by LekoArts — APPROVED** [review]

**Pull request merged**